### PR TITLE
Use dynamic dataset file name

### PR DESCRIPTION
**Pull request by mcornaton**

### CODE REVIEW

File: src/app.py [hunk 1]
```diff
@@ -5,35 +5,42 @@
...
+result = {"status_code": 0}
+dataset_exists = False
 
 # Eventually use some confidential assets
-if os.path.exists(iexec_in + '/data.json'):
-    with open(iexec_in + '/data.json', 'r') as json_file:
+if os.path.exists(dataset_path):
+    dataset_exists = True
+    with open(dataset_path, 'r') as json_file:
         secret = json.load(json_file)
```
Comment (jeremyjams) [inline src/app.py:19]: else write and quit asap?

Comment (mcornaton) [reply src/app.py:19]: That would duplicate the `with open...` blocks. I prefer current version.

Comment (jeremyjams) [reply src/app.py:19]: Oh missed the indentation on
`mailjet = Client(auth=(secret["api_key"], secret["api_secret"]), version='v3.1')`
!

File: src/app.py [hunk 2]
```diff
@@ -42,8 +49,8 @@
 print ("debug :" + str(result.json()))
 
 with open(iexec_out + '/result.txt', 'w+') as f:
-    if str(result.status_code) == "200":
-        f.write("email send successfully")
+    if not dataset_exists or not str(result.status_code) == "200":
```
Comment (jeremyjams) [inline src/app.py:52]:
```
if code == 200
else write failed
```
looks enough (initialized to 0 on L12)

Comment (mcornaton) [reply src/app.py]: Right, see https://github.com/iExecBlockchainComputing/confidential-email-sender/pull/3/commits/f12a596e3e77ba12978eab071beeaf6da5f59bfa.

Comment (jeremyjams) [reply src/app.py]: I don't see the added value of the `not`, but it is great otherwise, thanks :)

**Review by james-toussaint — APPROVED** [review]
Thanks :)

**Review by zguesmi — APPROVED** [review]
:+1:

**Pull request merged**